[#25] login

yeonsumia opened the pull request:
- send login confirm mail via POST /user/login
- verify token and return simpleResponse of user via GET /verify/login

## PR 타입
- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CI / CD
- [ ] 기타 설정

## 체크리스트
- [x] 작성한 코드가 프로젝트의 스타일과 맞습니다.
- [x] 코드의 이해에 필요한 주석을 작성했습니다.
- [ ] 수정 사항이 새로운 경고를 발생시키지 않습니다.
- [ ] 기존 및 신규 단위 테스트를 통과했습니다.
- [ ] (필요한 경우) 문서를 적절하게 수정했습니다.

yeonsumia committed:
[#25] login
- send login confirm mail via POST /user/login
- verify token and return simpleResponse of user via GET /verify/login

yeonsumia committed:
Merge branch 'master' into 25-login

WallabyG commented on waflog/src/main/kotlin/com/wafflestudio/waflog/global/auth/service/AuthService.kt:
```diff
@@ -61,15 +71,30 @@ class AuthService(
+            ?: throw TokenNotFoundException("잘못된 토큰")
+        val user = tokenUser.user
+        verificationTokenUserRepository.deleteById(tokenUser.id)
+        return UserDto.SimpleResponse(user)
```
이후에 JWT 기능 추가하면서 수정하시려는 것 같은데 맞나요?

wallabyg commented on waflog/src/main/kotlin/com/wafflestudio/waflog/global/auth/service/AuthService.kt:
```diff
@@ -61,15 +71,30 @@ class AuthService(
     }
 
-    private fun generateVerificationToken(email: String): String {
+    private fun generateNewVerificationToken(email: String): String {
```
new, exist 대신 signUp, signIn으로 구분해도 좋을 것 같습니다.

yeonsumia replied:
넵 수정하겠습니다.

wallabyg commented on waflog/src/main/kotlin/com/wafflestudio/waflog/global/auth/service/AuthService.kt:
```diff
@@ -61,15 +71,30 @@ class AuthService(
+            ?: throw TokenNotFoundException("잘못된 토큰")
+        val user = tokenUser.user
+        verificationTokenUserRepository.deleteById(tokenUser.id)
+        return UserDto.SimpleResponse(user)
```
이후에 JWT 기능 추가하면서 수정하시려는 것 같은데 맞나요?

yeonsumia replied:
넵 맞습니다.

yeonsumia committed:
[#46] jwt token authentication
- token login via GET auth/verify
- add columns(encoded token for password, role for authorities) on User model

yeonsumia committed:
[#46] modified security config

yeonsumia committed:
Merge pull request #50 from wafflestudio19-5/46-jwtTokenAuthentication

[#46] jwt token authentication

wallabyg reviewed (approved):
LGTM

yeonsumia committed:
[#25] modified function name

wallabyg commented:
혹시 핵심 로직을 수정하셔야 되나요?
그렇다면 approve 다시 풀어두겠습니다.

yeonsumia commented:
> 혹시 핵심 로직을 수정하셔야 되나요?
> 그렇다면 approve 다시 풀어두겠습니다.

아뇹 그냥 함수 명만 바꾸었습니다 ㅎㅎ,,
